src/test/java/com/getindata/kafka/connect/iceberg/sink/TestIcebergUtil.java

```diff
@@ -131,6 +131,73 @@ public void valuePayloadWithSchemaAsJsonNode() {
         assertFalse(deserializedSchema.has("schema"));
     }
 
+    @Test
+    public void coerceDebeziumTimeTypesDefaultBehavior(@TempDir Path localWarehouseDir)
+      throws JsonProcessingException {
+        IcebergSinkConfiguration config = TestConfig.builder()
+                .withLocalCatalog(localWarehouseDir)
+                .build();
+        config.configureChangeEvent();
+        IcebergChangeEvent e = new IcebergChangeEvent("test",
+                                          MAPPER.readTree(debeziumTimeCoercionSchema).get("payload"), null,
+                                          MAPPER.readTree(debeziumTimeCoercionSchema).get("schema"), null);
+        Schema schema = e.icebergSchema();
+        GenericRecord record = e.asIcebergRecord(schema);
+        String schemaString = schema.toString();
+        String recordString = record.toString();
+
+        assertTrue(schemaString.contains("ship_date: optional int (io.debezium.time.Date)"));
+        assertTrue(schemaString.contains("ship_timestamp: optional long (io.debezium.time.MicroTimestamp)"));
+        assertTrue(recordString.contains("77663"));
+        assertTrue(recordString.contains("6710075456016196"));
+    }
+
+    @Test
+    public void coerceDebeziumTimeTypesDisabledBehavior(@TempDir Path localWarehouseDir)
+      throws JsonProcessingException {
+        IcebergSinkConfiguration config = TestConfig.builder()
+                .withLocalCatalog(localWarehouseDir)
+                .withCustomProperty("coerce.debezium-date", "false")
+                .withCustomProperty("coerce.debezium-micro-timestamp", "false")
+                .build();
+        config.configureChangeEvent();
+        IcebergChangeEvent e = new IcebergChangeEvent("test",
+                                          MAPPER.readTree(debeziumTimeCoercionSchema).get("payload"), null,
+                                          MAPPER.readTree(debeziumTimeCoercionSchema).get("schema"), null);
+        Schema schema = e.icebergSchema();
+        GenericRecord record = e.asIcebergRecord(schema);
+        String schemaString = schema.toString();
+        String recordString = record.toString();
+
+        assertTrue(schemaString.contains("ship_date: optional int (io.debezium.time.Date)"));
+        assertTrue(schemaString.contains("ship_timestamp: optional long (io.debezium.time.MicroTimestamp)"));
+        assertTrue(recordString.contains("77663"));
+        assertTrue(recordString.contains("6710075456016196"));
+    }
```

Comment:
Nice coverage of all three states of the configuration: Defaulted, explicit disable, explicit enable 👍